compiler/rustc_hir_typeck/src/demand.rs

```diff
@@ -53,7 +57,8 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             || self.suggest_block_to_brackets_peeling_refs(err, expr, expr_ty, expected)
             || self.suggest_copied_or_cloned(err, expr, expr_ty, expected)
             || self.suggest_into(err, expr, expr_ty, expected)
-            || self.suggest_floating_point_literal(err, expr, expected);
+            || self.suggest_floating_point_literal(err, expr, expected)
+            || self.point_inference_types(err, expr);
```

Comment:
I'd prefer if we do something like:

```rust
let suggestion_applied = ...;

// If no suggestion is applied, point at where the inference happened.
if !suggestion_applied {
  self.point_at_inference_types(...);
}
```

That is, it feels somewhat out of place to chain this non-suggestion method with a bunch of other suggestion methods here.